homeassistant/components/alarm_control_panel/device_action.py

```diff
@@ -67,11 +70,12 @@ async def async_get_actions(
 
         supported_features = get_supported_features(hass, entry.entity_id)
 
-        base_action = {
+        base_action: dict = {
             CONF_DEVICE_ID: device_id,
             CONF_DOMAIN: DOMAIN,
             CONF_ENTITY_ID: entry.entity_id,
         }
+        async_set_entity_device_automation_metadata(base_action, entry)
```

Comment:
I guess it is the right level to do this because integrations know best if something is important or not. It also feels like we will replicate a lot of code and I wonder if we can add some defaults in the `device_automation` `get_device_actions` helper. Like `if CONF_ENTITY_ID in action: adjust_metadata`